src/xtgeo/grid3d/_gridprop_roxapi.py

```diff
@@ -176,16 +178,25 @@ def _store_in_roxar(self, pname, roxgrid, realisation, casting):  # pragma: no c
         rprop.code_names = self.codes.copy()
 
 
-def _fix_codes(codes):  # pragma: no cover
-    """Roxar may provide a code list with empty strings; fix this issue here."""
+def _fix_codes(self, codes):  # pragma: no cover
+    """Roxar may provide a code list with empty strings values, fix this issue here.
+
+    Roxar may also interpolate code values which are actually not present in the
+    property. Here, the presence of actual codes is also checked.
+    """
     newcodes = {}
+    codes_data = {val: str(val) for val in np.unique(self.get_active_npvalues1d())}
```

Comment:
Can be dropped

Reply:
Found that this construct was needed for the following cases:

rms_codes = {1: "", 2: "", 3: "", 4: ""}  In this case it may be that _only_ 1 and 4 actually exists, but RMS interpolates. 
XTGeo want: {1: "1", 4: "4"}

rms_codes = {1: "SAND", 2: "", 3: "", 4: "SHALE"}. Only 1 and 4 are given proper names, but we are _not sure_ if 2 and/or 3 also exists, If e.g 3 actually is present also, we want: {1: "SAND", 3: "3", 4: "SHALE"}. More extreme, it may be possible that  although 4: "SHALE" is named, it is not present, hence result should be: {1: "SAND", 3: "3"}

Reply:
Yes, I understand the problem. My suggestion was simply to do the check against the active values directly, instead of creating a dictionary to check against. But minor detail, I'll approve :)

Reply:
Add test code for this?